### DISCUSSION TITLE
tonic client server side streaming call blocks when interfacing with csharp grpc server

### DISCUSSION THREAD

**Discussion by dougcooper** [hyperium/tonic#1233] [Q&A]
I've tested this against a rust service and csharp service that effectively blocks forever before writing to the stream and am seeing different client behavior as follows:

* In the case of the rust service my tonic client gets a valid stream and i can call .next() to wait for a message.
* In the case of the csharp service my tonic client never returns from the initial api call and so i never have a valid stream to iterate over

In both cases i sleep in order to simulate a stream that doesnt have any data immediately available.

What Im trying to do is make several streaming based api calls and then use tokio::select! to multiplex the responses in an infinite loop or until a stream gets dropped.

Please let me know if i can clarify...cheers

**Comment by LucioFranco** [comment]
What sleep are you using in rust?

**Reply by dougcooper** [reply]
`tokio::time::sleep(Duration::from_secs(3600)).await;`

**Reply by LucioFranco** [reply]
Its hard to say without the full code context unfortunately.

**Reply by dougcooper** [reply]
I recreated the issue using trivially modified off the shelf examples and a python service. here's the setup:

## python server (rust client does not return a stream when no data immediately available)

```
git clone https://github.com/dougcooper/grpc-test && cd grpc-test/examples/python
python3 -m venv .venv
. .venv/bin/activate
pip3 install grpcio-tools grpcio
make
python3 hellostreamingworld/async_greeter_server.py
```

## rust server (rust client gets  a stream as expected with no data immediately available)

```
git clone https://github.com/dougcooper/tonic-test && cd tonic-test/examples
cargo run --bin streaming-server
```

## rust client

```
cargo run --bin streaming-client
```

## expected behavior

after running the server then launching the client i **should** see a client log message that says "we have a stream" but not receive any data due to server side sleep.

**Reply by dougcooper** [reply]
i also tested the async_greeter_client.py against the async_greeter_server.py and it works as expected.

**Reply by LucioFranco** [reply]
If you think its a proper bug then I would open an issue with a minimal reproduction.
